env/dev/secretsmanager.tf

```diff
@@ -0,0 +1,88 @@
+# create the secretsmanager secret
+resource "aws_secretsmanager_secret" "sm_secret" {
+  name = "${var.app}-${var.environment}"
+  tags = "${var.tags}"
+  policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "DenyWriteToAllExceptSAMLUsers",
+      "Effect": "Deny",
+      "Principal": "*",
+      "Action": [
+        "secretsmanager:CancelRotateSecret",
+        "secretsmanager:CreateSecret",
+        "secretsmanager:DeleteResourcePolicy",
+        "secretsmanager:DeleteSecret",
+        "secretsmanager:GetResourcePolicy",
+        "secretsmanager:PutResourcePolicy",
+        "secretsmanager:PutSecretValue",
+        "secretsmanager:RestoreSecret",
+        "secretsmanager:RotateSecret",
+        "secretsmanager:TagResource",
+        "secretsmanager:UntagResource",
+        "secretsmanager:UpdateSecret",
+        "secretsmanager:UpdateSecretVersionStage"
+      ],
+      "Resource": [
+        "arn:aws:secretsmanager:${var.region}:${data.aws_caller_identity.current.account_id}:secret:${var.app}-${var.environment}-*"
+      ],
+      "Condition": {
+        "StringNotLike": {
+          "aws:arn": [
+            "arn:aws:sts::${data.aws_caller_identity.current.account_id}:assumed-role/${var.saml_role}/me@example.com"
+          ]
```

Comment:
Wondering if we should replace `me@example.com` with a terraform variable to make it more obvious that at least one saml user needs to be added.  I forgot to change this the first time while testing and locked myself out.

Reply:
Yes.  I was thinking that too.  Also means I can change it in one place.

Reply:
We could doing something like this, to pass them as a list... 
```
  secrets_saml_users = [
    "me@example.com,
    "you@example.com,
  ]
```

example here: https://github.com/turnerlabs/terraform-s3-employee/blob/master/main.tf#L62-L71

Reply:
Although, making it a variable means we probably have to make it a template so that we can interpolate the account and all that.   Thats painful if you have to also include roles in your list.